src/torrent.cpp

```diff
@@ -1293,6 +1293,12 @@ bool is_downloading_state(int const st)
 		return m_ses.disk_thread().get_torrent(m_storage);
 	}
 
+    void torrent::set_sequential_start(piece_index_t piece)
+    {
+        if (!m_picker) return;
+        m_picker->set_sequential_start(piece);
```

Comment:
if the torrent doesn't have a piece picker at this point (say, because it's a magnet link that hasn't received metadata), this setting will silently be ignored. That doesn't seem right.

Reply:
@arvidn This is all very right. There is no way in the library to check if m_picker is created. So sometimes calls happen at the wrong time. In order not to get a crash - this check.

Reply:
well, it's not documented to have this limitation

Reply:
@arvidn Yes. You're right. We should document that. Where to document this?

Reply:
add or update comments above public functions in public headers. In this case, `include/libtorrent/torrent_handle.hpp`. It might also make sense to update the piece picker documentation in `docs/manual.rst`

Reply:
@arvidn Will you add the documentation yourself to docs/manual.rst ? Okay? 
I tried to describe set_sequential_start correctly in torrent_handle.hpp. You can take a look, adjust and copy it where you need it.